test/spec/modules/greenbidsBidAdapter_specs.js

```diff
@@ -0,0 +1,964 @@
+import { expect } from 'chai';
+import { newBidder } from 'src/adapters/bidderFactory.js';
+import { spec } from 'modules/greenbidsBidAdapter.js';
+const ENDPOINT = 'https://d.greenbids.ai/hb/bid-request';
+const AD_SCRIPT = '<script type="text/javascript" class="greenbids" async="true" src="https://greenbids.ai/settings"></script>"';
+
+describe('greenbidsBidAdapter', () => {
+  const adapter = newBidder(spec);
+  let sandbox;
+
+  beforeEach(function () {
+    sandbox = sinon.sandbox.create();
+  });
+
+  afterEach(function () {
+    sandbox.restore();
+  });
+
+  describe('inherited functions', () => {
+    it('exists and is a function', () => {
+      expect(adapter.callBids).to.exist.and.to.be.a('function');
+    });
+  });
+
+  describe('isBidRequestValid', function () {
+    let bid = {
+      'bidder': 'greenbids',
+      'params': {
+        'placementId': 4242
+      },
+      'adUnitCode': 'adunit-code',
+      'sizes': [[300, 250], [300, 600]],
+      'bidId': '30b31c1838de1e',
+      'bidderRequestId': '22edbae2733bf6',
+      'auctionId': '1d1a030790a475',
+      'creativeId': 'er2ee',
+    };
+
+    it('should return true when required params found', function () {
+      expect(spec.isBidRequestValid(bid)).to.equal(true);
+    });
+
+    let bidNonGBCompatible = {
+      'bidder': 'greenbids',
+    };
+
+    it('should return false when required params are not found', function () {
+      expect(spec.isBidRequestValid(bidNonGBCompatible)).to.equal(false);
+    });
+
+    let bidNonGBCompatible2 = {
+      'bidder': 'greenbids',
+      'params': {
+        'placementId': 'toto'
+      },
```

Comment:
suggestion: instead of having `bidNonGBCompatible2` could it be better (maybe it's not javascript-ist) to declare the variables inside the `it` block ?